spark/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSQLConf.scala

```diff
@@ -1329,6 +1329,42 @@ trait DeltaSQLConfBase {
       .booleanConf
       .createWithDefault(false)
 
+
+  /////////////////////
+  // Optimized Write
+  /////////////////////
+
+  val DELTA_OPTIMIZE_WRITE_ENABLED =
+    buildConf("optimizeWrite.enabled")
+      .doc("Whether to optimize writes made into Delta tables from this session.")
+      .booleanConf
+      .createOptional
+
+  val DELTA_OPTIMIZE_WRITE_SHUFFLE_BLOCKS =
+    buildConf("optimizeWrite.numShuffleBlocks")
+      .internal()
+      .doc("Maximum number of shuffle blocks to target for the adaptive shuffle " +
+        "in optimized writes.")
+      .intConf
+      .createWithDefault(50000000)
```

Comment:
is there a reason for choosing these defaults

Reply:
As explained in the config doc, `optimizeWrite.maxShufflePartitions` must be not be larger than `spark.shuffle.minNumPartitionsToHighlyCompress`, which is 2000 by default. And then `optimizeWrite.numShuffleBlocks` is set high enough to produce sufficient number of partitions (while still being limited by `optimizeWrite.maxShufflePartitions`).

Reply:
got it